services/blockstorage/sync/state_processing_blocks.go

```diff
@@ -0,0 +1,69 @@
+package sync
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+)
+
+type processingBlocksState struct {
+	blocks  *gossipmessages.BlockSyncResponseMessage
```

Comment:
it's like just a *bit* ugly that you hold the gossip message here, but is it better to duplicate the struct? I truly don't know